src/PWFusion_MAX31856.cpp

```diff
@@ -179,22 +187,28 @@ void MAX31856::config(Tc_Type TC_TYPE, uint8_t FILT_FREQ, uint8_t AVG_MODE, Max3
 
 void MAX31856::sample()
 {
+   uint16_t rawCJReg;
+   uint32_t rawTCReg;
    // Start by reading SR for any faults, exit if faults present, though some
    // faults could potentially be dealt with
    status = readByte(REG_SR);
    
    // Read Cold Jcn temperature (2 registers)
-   rawCJTemp  = (int16_t)(uint16_t)readByte(REG_CJTH) << 8; // MSB, left shift 8
-   rawCJTemp |= (int16_t)(uint16_t)readByte(REG_CJTL);              // LSB read
-   // now save sign, shift right 2 to align to type (see datasheet, pg 24 for reg packing)
-	rawCJTemp >>= 2;
+   rawCJReg  = (int16_t)(uint16_t)readByte(REG_CJTH) << 8; // MSB, left shift 8
+   rawCJReg |= (int16_t)(uint16_t)readByte(REG_CJTL);              // LSB read
+
+   // convert to 16-bit sign-magnitude value
+   rawCJReg = (rawCJReg >> 2) & 0b1001111111111111;
+   rawCJTemp = smtotc_16(rawCJReg);

```

Comment:
What do you think of this?  The smtotc_16 implementation looked right, but is hard to follow all the bit math.

```suggestion
	
   // convert sign-magnitude value to 2's complement.   If bit 0x2000 is set this is a negative number, 
   //convert all the bits to the right of the sign bit to 2s complement
   if (rawCJReg & 0x2000)
   {
      rawCJReg = -(rawCJReg & 0x1FFF);
   }

```